merlin-worker/src/main/java/gov/nasa/jpl/aerie/merlin/worker/ListenSimulationCapability.java

```diff
@@ -0,0 +1,60 @@
+package gov.nasa.jpl.aerie.merlin.worker;
+
+import com.impossibl.postgres.api.jdbc.PGConnection;
+import com.impossibl.postgres.api.jdbc.PGNotificationListener;
+import gov.nasa.jpl.aerie.merlin.server.remotes.postgres.DatabaseException;
+import gov.nasa.jpl.aerie.merlin.worker.postgres.ListenSimulationStatusAction;
+import gov.nasa.jpl.aerie.merlin.worker.postgres.PostgresSimulationNotificationPayload;
+
+import javax.json.Json;
+import javax.sql.DataSource;
+import java.io.StringReader;
+import java.sql.SQLException;
+import java.util.concurrent.BlockingQueue;
+
+import static gov.nasa.jpl.aerie.merlin.worker.postgres.PostgresNotificationJsonParsers.postgresSimulationNotificationP;
+
+public class ListenSimulationCapability {
+  private final DataSource dataSource;
+  private final BlockingQueue<PostgresSimulationNotificationPayload> notificationQueue;
+
+  public ListenSimulationCapability(
+      final DataSource dataSource,
+      final BlockingQueue<PostgresSimulationNotificationPayload> notificationQueue) {
+    this.dataSource = dataSource;
+    this.notificationQueue = notificationQueue;
+  }
+
+  public void registerListener() throws SQLException {
+    PGConnection connection = this.dataSource.getConnection().unwrap(PGConnection.class);
+    connection.addNotificationListener(new SimulationNotificationListener());
+
+    try (final var listenSimulationStatusAction = new ListenSimulationStatusAction(connection)) {
+       listenSimulationStatusAction.apply();
+    } catch (final SQLException ex) {
+      throw new DatabaseException("Failed to register as LISTEN to postgres database.", ex);
+    }
+  }
+
+  private class SimulationNotificationListener implements PGNotificationListener {
+
+    @Override
+    public void notification(int processId, String channelName, String payload){
+      System.out.println("Received PSQL Notification: " + processId + ", " + channelName + ", " + payload);
+      final var jsonValue = Json.createReader(new StringReader(payload)).readValue();
+      final var notificationPayload = postgresSimulationNotificationP
+          .parse(jsonValue)
+          .getSuccessOrThrow();
+      try {
+        notificationQueue.put(notificationPayload);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

Comment:
🔻 Flagging this catch block - we should either rethrow some kind of an exception, or write a comment explaining why we're suppressing it.

Reply:
Have resolved this by the resetting the interruption. Reading in "[Java Concurrency In Practice](https://jcip.net/)", they recommend resetting the interrupt when you are in a context that prevent you from taking action. We have no ability to take action within this listener callback.